platforms/webextension/web-discovery-project/doublefetch.es

```diff
@@ -4,12 +4,37 @@
 
 import { parse, equals as urlEquals } from "../../core/url";
 import { clearTimeout, setTimeout } from "../../core/timers";
+import { getDomainWithoutSuffix } from "../../core/tlds";
 import { fetch, AbortController, Headers } from "../fetch";
 
 // There needs to proper implementation, to avoid cases like:
 // 1. Downloading streams.
 // 2. Origin in web-extension.
-export function getRequest(url, headers, extra) {
+export function getRequest(url, headers) {
+  if (
+    chrome?.webDiscovery?.retrieveBackupResults !== undefined &&
+    getDomainWithoutSuffix(request.url) == "google" &&
+    request.url.includes("/search?")
+  ) {
+    return new Promise((resolve, reject) => {
+      chrome.webDiscovery.retrieveBackupResults(url, (response, err) => {
```

Comment:
unfortunately `err` does not exist. we should check `chrome.runtime.lastError` instead